### ISSUE TITLE
`ai/functiongemma` tool calls are not parsed into `tool_calls`

### ISSUE THREAD

**Issue by yohasebe** [docker/model-runner#1045]
### Summary

`ai/functiongemma` emits well-formed tool-call tokens in its own format, but Docker Model Runner does not parse them:

- `tool_calls` stays `null` and the raw `<start_function_call>…<end_function_call>` text is returned in `message.content`;
- `<end_function_call>` is not treated as a stop token, so the call repeats until `max_tokens` and `finish_reason` is `"length"`.

There is no error. To a client, the response looks like the model simply did not use the tool.

### Steps to reproduce

```bash
docker model pull ai/functiongemma

curl -s http://localhost:12434/engines/v1/chat/completions \
  -H "Content-Type: application/json" \
  -d '{
    "model": "ai/functiongemma",
    "max_tokens": 400,
    "messages": [{"role":"user","content":"What is the current time? Use the get_time tool."}],
    "tools": [{"type":"function","function":{
      "name":"get_time","description":"Get the current time",
      "parameters":{"type":"object","properties":{}}}}]
  }' | jq '.choices[0]'
```

**Actual**

```
tool_calls    : null
finish_reason : "length"
content       : "<start_function_call>call:get_time{}<end_function_call>call:get_time{}<end_function_call>call:get_ti…"
```

**Expected**

`tool_calls` populated, `finish_reason: "tool_calls"`, and no call tokens in `content`. For comparison, an equivalent request to Ollama returns:

```json
{"tool_calls": [{"id": "call_o3fkhnnc", "function": {"name": "get_time", "arguments": {}}}]}
```

with `content` empty.

### Suggested resolution

Parse the `<start_function_call>…<end_function_call>` format into `tool_calls` and treat `<end_function_call>` as a stop token.

Related: #1046. `/api/show` exposes no `capabilities`, so a client also cannot check for tool support before making the request.

### Environment

- Model Runner client `v1.2.6`, Docker Desktop engine `29.7.2`
- Backend `llama.cpp b9879-metal`, macOS, Apple Silicon
- Model: `ai/functiongemma`

**Comment by houleixx2** [comment]
A useful parser regression trace would preserve the response from the first `<start_function_call>` through the first `<end_function_call>`, before the repeated text reaches `max_tokens`. Alongside that, record the effective model metadata/chat template and compare with the same request through Ollama. This can distinguish “model runner never enabled the functiongemma parser” from “the parser ran but did not recognize the emitted variant.”

I maintain ccglass, which can inspect the OpenAI-compatible request and streamed/non-streamed response and export a redacted event timeline. It cannot see raw logits or internal llama.cpp template state, but it can provide a stable wire fixture showing the exact marker bytes, repetition point, `finish_reason`, and missing `tool_calls` field for validating a fix.

Project: https://github.com/jianshuo/ccglass
